### ISSUE TITLE
Setting accessibleRole="none" on a `Text` component doesn't seem to affect accessibility behavior

### ISSUE THREAD

**Issue by chrisglein** [microsoft/react-native-windows#11376]
### Problem Description

I'm not actually sure what the accessibility behavior should be here. But it seemed wrong that setting accessibilityRole="none" had no effect.
```
<Text>⚙️</Text>
```
![image](https://user-images.githubusercontent.com/26607885/225158183-f2249c46-a097-4160-bde8-2cd7f2d922cf.png)

```
<Text accessibilityRole="none">⚙️</Text>
```
![image](https://user-images.githubusercontent.com/26607885/225158001-d95cd322-d6e1-4a13-952e-c10aeed5a359.png)

```
<Text accessible={false}>⚙️</Text>
```
![image](https://user-images.githubusercontent.com/26607885/225158367-beb2d3e5-c3d5-429c-b9e5-a6c437cb556a.png)

### Steps To Reproduce

See code above

### Expected Results

Element is hidden in some way when role is set?

### CLI version

10.0.0

### Environment

```markdown
System:
    OS: Windows 10 10.0.23415
    CPU: (24) x64 AMD Ryzen Threadripper PRO 3945WX 12-Cores
    Memory: 38.74 GB / 63.86 GB
  Binaries:
    Node: 16.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 9.4.1 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0, 10.0.22000.0, 10.0.22621.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 17.4.33403.182 (Visual Studio Community 2022), 16.11.33328.57 (Visual Studio Community 2019)
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0
    react-native: 0.71.0 => 0.71.0
    react-native-windows: 0.71.1 => 0.71.1
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

10.0.19041

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug

**Comment by chrisglein** [comment]
Duplicate of #8191
